### PR TITLE
[FEAT] NoOp adapters for load testing

### PR DESCRIPTION
**Pull request by harini-venkataraman**
## What
PR to add support for NoOp adapters, which adds configured delay time, and without any specific function.
...

## Why
To avoid suspects on throttle due to external communication and arrive at definitive results on load testing.
...

## How
Adding NoOp adapters for LLM, Embedding, vectorDb and text extractors.
...

## Relevant Docs
Not applicable.
...

## Related Issues or PRs
Not applicable.
...

## Dependencies Versions / Env Variables
SDK version bumped to 0.51.0
...

## Notes on Testing
Flow :
1. Add the NoOp adapters in the settings
2. Configure in the Profile managers
3. Use them for load testing.
...

## Screenshots
Screenshot from 2024-10-10 17-00-30

Screenshot from 2024-10-10 16-59-50

Screenshot from 2024-10-10 16-57-31

...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      "type": "string",
+      "title": "Name",
+      "default": "",
+      "description": "Provide a unique name for this No Op adapter instance. Example: no-op-instance-1"
+    },
+    "wait_time": {
+      "type": "number",
+      "title": "Wait time",
+      "default": "0",
+      "description": "Provide the wait time."
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:19]:
```suggestion
      "description": "Provide the time to wait before returning the response"
```

File: src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm_adapter.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    def test_connection(self) -> bool:
+        llm = self.get_llm_instance()
+        if not llm:
+            return False
+        response = llm.complete(
+            "The capital of Tamilnadu is ",
+            temperature=0.003,
+        )
+        response_lower_case: str = response.text.lower()
+        if response_lower_case:
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm_adapter.py:58]: @harini-venkataraman why do we perform this check? Shouldn't we return `True` always?

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    def test_connection(self) -> bool:
+        time.sleep(self._config.get("wait_time"))
+        return True
+
+    def close(self, **kwargs: Any) -> None:
+        pass
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        time.sleep(self._config.get("wait_time"))
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py:77]: NIT: This would imply that during indexing we have 2 `sleep()`s when we delete and add the nodes and at other times we'll `sleep()` once. I hope this behaviour is intended and conveyed

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                VectorDbConstants.DEFAULT_EMBEDDING_SIZE,
+            )
+            self._collection_name = VectorDBHelper.get_collection_name(
+                self._config.get(VectorDbConstants.VECTOR_DB_NAME),
+                dimension,
+            )
+            vector_db: VectorStore = NoOpVectorDB(
+                dim=dimension, wait_time=self._config.get(VectorDbConstants.WAIT_TIME)
+            )
+            if vector_db is not None:
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py:63]: @harini-venkataraman is this check needed? Shouldn't we just return the client?

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "wait_time"
+  ],
+  "properties": {
+    "adapter_name": {
+      "type": "string",
+      "title": "Name",
+      "default": "",
+      "description": "Provide a unique name for this No Op adapter instance. Example: no-op-instance-1"
+    },
+    "wait_time": {
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @harini-venkataraman I hope this works as expected for floats as well

File: src/unstract/sdk/index.py [hunk 6]
```diff
@@ -290,9 +293,18 @@ def index(
...
             if not extracted_text:
                 raise IndexingError("No text available to index")
 
+            if isinstance(
+                vector_db.get_vector_db(
+                    adapter_instance_id=vector_db_instance_id, embedding_dimension=1
+                ),
+                (NoOpVectorDB),
+            ):
+                return doc_id
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/index.py:306]: @harini-venkataraman why do we need this check? Also why do we wait until extraction to return something we calculate before the extraction?

File: src/unstract/sdk/llm.py [hunk 7]
```diff
@@ -251,6 +252,11 @@ def run_completion(
                 raise RateLimitError(msg)
             raise LLMError(msg) from e
 
+    def is_noOp(self, llm: LlamaIndexLLM) -> bool:
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/llm.py:255]: @harini-venkataraman why is this necessary in the base class? Its similar to defining a function to check if its of type `OpenAILLM`

**Review by chandrasekharan-zipstack — COMMENTED** [review]
Haven't duplicated the comments to all adapters, please extend it to the other places where it might apply as well

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/no_op_embedding_adapter.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        return schema
+
+    def get_embedding_instance(self) -> BaseEmbedding:
+        embedding: BaseEmbedding = NoOpEmbedding(
+            embed_dim=1, wait_time=self.config.get("wait_time")
+        )
+        return embedding
+
+    def test_connection(self) -> bool:
+        time.sleep(self.config.get("wait_time"))
```
[the remark this answers is not in the document]
Comment (Deepak-Kesavan) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/no_op_embedding_adapter.py:52]: Yes. We can remove this.

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json (hunk 1 above)
Comment (Deepak-Kesavan) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:19]: @harini-venkataraman Please mention the unit as well in the title and description. In this case it is seconds.

File: src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+                delta=response_text,
+            )
+
+        time.sleep(self.wait_time)
+
+        return gen_response()
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        return LLMMetadata(num_output=-1)
```
[the remark this answers is not in the document]
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm.py:57]: This method ideally, is used to return LLM metadata, which has context window,  model name. Need to override here since we extend CustomLLM

File: src/unstract/sdk/llm.py (hunk 7 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/llm.py:255]: @chandrasekharan-zipstack This method is introduced since we needed to identify the NoOp adapter to differentiate JSON type response for enforce type like JSON, table and record.

File: src/unstract/sdk/index.py (hunk 6 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/index.py:306]: @chandrasekharan-zipstack 
1. To answer to first question, right after this part of snippet, we do addition of nodes to vectorDB which has to skipped for NoOps. Hence the check. 
2. In the indexing, initialisation of embedding and its generation and x2Text NoOp is to be invoked to retian the flow as such. Hence added after the extraction.

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json (hunk 5 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @chandrasekharan-zipstack I doubt if we need to support float for wait time. 
@hari-kuriakose can you please help here?

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/no_op_embedding_adapter.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/no_op_embedding_adapter.py:52]: Yeah, this should have not been added. Thanks for pointing it. Addressed.

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py (hunk 3 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py:77]: @chandrasekharan-zipstack Actually in the flow, we skip the part where we delete the nodes in case of NoOp and hence it will be just once.

File: src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    ) -> CompletionResponseGen:
+
+        def gen_response() -> CompletionResponseGen:
+            response_text = self._generate_text()
+            yield CompletionResponse(
+                text=response_text,
+                delta=response_text,
+            )
+
+        time.sleep(self.wait_time)
```
[the remark this answers is not in the document]
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm.py:51]: @chandrasekharan-zipstack It will be consuming a lot of time as we don't have a control over the number of returned stream objects. Consequently, there are even risks of going into a longer sleep when there are many tokens returned if we sleep everytime we stream.

File: src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm.py:57]: @harini-venkataraman please add a comment for the same to not lose this context

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py (hunk 3 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py:77]: Okay, make sure to mention it in the description of the adapter that we sleep for such operations

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json (hunk 5 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @harini-venkataraman if its a simple change / just works - it'll be a good to have. Do we parse to int anywhere?

Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @harini-venkataraman No need to ensure support of float for now.

File: src/unstract/sdk/llm.py (hunk 7 above)
Comment (hari-kuriakose) [inline src/unstract/sdk/llm.py:255]: @harini-venkataraman Please rename function to `is_no_op`.

File: src/unstract/sdk/llm.py [hunk 11]
```diff
@@ -251,6 +252,11 @@ def run_completion(
...
+        if isinstance(llm, NoOpLLM):
+            return True
+        return False
+
```
Comment (hari-kuriakose) [inline src/unstract/sdk/llm.py:259]:
```suggestion
        return isinstance(llm, NoOpLLM)
```

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        """Query vector store."""
+        node1 = TextNode(text="This is a dummy document.", id_="1")
+        similarity_scores = [0.9]
+        embeddings = [["test"]]  # Dummy embeddings for each node
+
+        query_result = VectorStoreQueryResult(
+            nodes=[node1], similarities=similarity_scores, ids=embeddings
+        )
+        time.sleep(self.wait_time)
+        return query_result
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb.py:32]: @harini-venkataraman Don't we need add() also here to dummify add?

Comment (gaya3-zipstack) [reply src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb.py:32]: I do see that for simplicity we have skipped a chunk of flows. . Pls cross check once to make sure if we have dummified the required core functions from llama-index that are being used in our flows.

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json (hunk 5 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @chandrasekharan-zipstack We dont explicitly parse, but have static typing.
I just noticed, sleep itself takes an argument in float. Changing the typing to float.

Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @chandrasekharan-zipstack We dont explicitly parse, but have static typing.
I just noticed, sleep itself takes an argument in float. Changing the typing to float.

File: src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+        input_file_path: str,
+        output_file_path: Optional[str] = None,
+        **kwargs: dict[Any, Any],
+    ) -> TextExtractionResult:
+        extracted_text: str = (
+            "This is a No Op x2text adapter response."
+            " This is a sample response and intended for testing \f"
+        )
+        time.sleep(self.config.get("wait_time"))
+        return TextExtractionResult(extracted_text=extracted_text)
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py:51]: Pls make sure that choosing not to write to output_file_path etc will not cause any logical issues in any flow.

File: src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm_adapter.py (hunk 2 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/llm/no_op_llm/src/no_op_llm_adapter.py]: Addressed

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json (hunk 5 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:15]: @hari-kuriakose Mere changing typing works good for float as well. JSON Schema already has `number` datatype.

File: src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py (hunk 13 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py:51]: @gaya3-zipstack Preferred not to write to file since didn't want to junk the storage with the sample response text file. 
The only place where we use it is when we switch to `Raw view` in document viewer. When NoOp x2text is chosen, it displays a message of "File not found", it is explanatory when a NoOp is used.
@hari-kuriakose Can you please help here?

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb.py (hunk 12 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb.py:32]: @gaya3-zipstack In indexing flow, we have skipped the snippet where the actual addition of nodes happen. It does not trigger indexing and completes with an assumption that it is already indexed. 
While retrieval, the overridden `query` method caters to the dummy nodes. 
If needed, wee can get on a quick call to discuss this.

Comment (gaya3-zipstack) [reply src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb.py:32]: Got it @harini-venkataraman

File: src/unstract/sdk/llm.py (hunk 7 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/llm.py]: @harini-venkataraman its not a good practice to include such a method in the base class implementation. Do you have ideas on how we can avoid it?

File: src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py (hunk 13 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py:54]: @gaya3-zipstack Like we discussed, this will be needed when summary is enabled. To retain the flow, we might need to write to the path. Making the change, thanks for pointing it out.

Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/x2text/no_op_x2text/src/no_op_x2text_adapter.py:54]: @gaya3-zipstack Like we discussed, this will be needed when summary is enabled. To retain the flow, we might need to write to the path. Making the change, thanks for pointing it out.

File: src/unstract/sdk/llm.py (hunk 7 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/llm.py]: @chandrasekharan-zipstack I understand that checking a specific adapter instance is a very targeted approach in the base class. To enforce types like Table, we utilize the details returned for subsequent operations, which may ultimately raise an exception. Instead, we might need to  handle this by checking whether we are using the no-op adapter and return early if so.

While this does not necessarily require the inclusion of a new method in the base class, we can perform the necessary checks using existing getters within the backend/prompt_service. It seems more appropriate to keep the adapter logic within the LLM's base class, rather than integrating it into this flow. For reference, please see the relevant code [here](https://github.com/Zipstack/unstract-cloud/blob/4115532bb9a17683e30eb17b81c2a69f7f679384/prompt_service/plugins/table-extractor/src/base.py#L41).
CC : @hari-kuriakose

**Review by gaya3-zipstack — APPROVED** [review]
My concerns seem addressed. Overall looks safe for dummy adapters.

**Review by Deepak-Kesavan — APPROVED** [review]

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+                self._client = vector_db.client
+            return vector_db
+        except Exception as e:
+            raise AdapterError(str(e))
+
+    def test_connection(self) -> bool:
+        time.sleep(self._config.get("wait_time"))
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/no_op_vectordb_adapter.py]:
```suggestion
```
@harini-venkataraman please remove such sleeps in other adapters as well

File: src/unstract/sdk/index.py (hunk 6 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/index.py:306]: @harini-venkataraman why do you feel that addition of the nodes to vector DB has to be skipped for no ops?
- In my opinion we could have fractional timeouts and add a sleep for all vector DB ops. This will allow us to avoid checking for `no op` adapter type in other parts of the code

**Review by chandrasekharan-zipstack — APPROVED** [review]
LGTM overall.
Have 2 minor concerns around checking for no-op type in places where adapters are used - I hope there's general consensus around these @hari-kuriakose @harini-venkataraman @gaya3-zipstack @Deepak-Kesavan 
1. [Checking it in index before adding nodes to vectorDB](https://github.com/Zipstack/unstract-sdk/pull/109/files#r1796155556)
1. Adding a method to check for this instance in the base class - related to table extractor

File: src/unstract/sdk/index.py (hunk 6 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/index.py:306]: @chandrasekharan-zipstack  The actual addition of nodes takes place in[ this part](https://github.com/Zipstack/unstract-sdk/blob/c48e21f45b92a158ab22d770a67afb5abdc0fe60/src/unstract/sdk/index.py#L310) of the code which need not be invoked for NoOps. This is because the base class invokes LLama index class method by passing the NoOp instance and this instance does not have any control.
Also, logically we dont skip for another places except for enforce types.

File: src/unstract/sdk/llm.py (hunk 7 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/llm.py]: @chandrasekharan-zipstack  Considering the NoOps, added a check in the flow and removed from here. Thanks for the suggestion.

File: src/unstract/sdk/adapters/llm/no_op_llm/src/static/json_schema.json [hunk 15]
```diff
@@ -0,0 +1,22 @@
+{
+  "title": "NoOp LLM Adapter",
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/llm/no_op_llm/src/static/json_schema.json:2]: @harini-venkataraman Please rename to just "No-op".

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/static/json_schema.json [hunk 16]
```diff
@@ -0,0 +1,22 @@
+{
+  "title": "NoOp VectorDB Adapter",
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/vectordb/no_op_vectordb/src/static/json_schema.json:2]: @harini-venkataraman Please rename to just "No-op".

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json [hunk 17]
```diff
@@ -0,0 +1,22 @@
+{
+  "title": "NoOp Embedding Adapter",
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json:2]: @harini-venkataraman Please rename to just "No-op".

File: src/unstract/sdk/adapters/embedding/no_op_embedding/README.md
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/embedding/no_op_embedding/README.md:1]: @harini-venkataraman Please rename folder to just "no_op".

File: src/unstract/sdk/adapters/llm/no_op_llm/README.md
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/llm/no_op_llm/README.md:1]: @harini-venkataraman Please rename folder to just "no_op".

File: src/unstract/sdk/adapters/vectordb/no_op_vectordb/README.md
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/vectordb/no_op_vectordb/README.md:1]: @harini-venkataraman Please rename folder to just "no_op".

File: src/unstract/sdk/adapters/x2text/no_op_x2text/README.md
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/x2text/no_op_x2text/README.md:1]: @harini-venkataraman Please rename folder to just "no_op".

File: src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+        input_file_path: str,
+        output_file_path: Optional[str] = None,
+        **kwargs: dict[Any, Any],
+    ) -> TextExtractionResult:
+        extracted_text: str = (
+            "This is a No Op x2text adapter response."
+            " This is a sample response and intended for testing \f"
+        )
+        time.sleep(self.config.get("wait_time"))
+        if output_file_path:
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py:51]: @gaya3-zipstack This location also need to switch to new FS API interface that we are bringing in, if not considered already.

File: src/unstract/sdk/index.py (hunk 6 above)
Comment (hari-kuriakose) [reply src/unstract/sdk/index.py:304]: @harini-venkataraman Ideally agree here that we should not need any explicit check whether it is a No-op adapter.

However, I understand that we cannot efficiently do this currently because it will need some refactoring to accommodate for the blackbox handling of LlamaIndex callback manager etc.

Hence request you to add a dev comment here surmising these points for future reference.

File: src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py [hunk 19]
```diff
@@ -0,0 +1,32 @@
+import time
+
+from llama_index.core.schema import TextNode
+from llama_index.core.vector_stores.types import VectorStore, VectorStoreQueryResult
+
+
+class NoOpVectorDB(VectorStore):
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py:7]: @harini-venkataraman Better to rename this to `NoOpCustomVectorDB`.

Similarly we can do in other places.

File: src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb_adapter.py [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+from llama_index.core.vector_stores.types import VectorStore
+
+from unstract.sdk.adapters.exceptions import AdapterError
+from unstract.sdk.adapters.vectordb.constants import VectorDbConstants
+from unstract.sdk.adapters.vectordb.helper import VectorDBHelper
+from unstract.sdk.adapters.vectordb.no_op.src.no_op_vectordb import NoOpVectorDB
+from unstract.sdk.adapters.vectordb.vectordb_adapter import VectorDBAdapter
+
+
+class NoOpVectorDBAdapter(VectorDBAdapter):
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb_adapter.py:15]: @harini-venkataraman Better to rename this to `NoOpVectorDB`.

Similarly we can do so in other places.

File: src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+import time
+from typing import Any, Optional
+
+from unstract.sdk.adapters.x2text.dto import TextExtractionResult
+from unstract.sdk.adapters.x2text.x2text_adapter import X2TextAdapter
+
+logger = logging.getLogger(__name__)
+
+
+class NoOpX2TextAdapter(X2TextAdapter):
```
Comment (hari-kuriakose) [inline src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py:12]: @harini-venkataraman We could rename this to `NoOpX2Text`.

File: src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py (hunk 18 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py:51]: @gaya3-zipstack Hope you have this in track, resolving it here.

Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py:51]: @gaya3-zipstack Hope you have this in track, resolving it here.

File: src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+            )
+            vector_db: VectorStore = NoOpCustomVectorDB(
+                dim=dimension, wait_time=self._config.get(VectorDbConstants.WAIT_TIME)
+            )
+            self._client = vector_db.client
+            return vector_db
+        except Exception as e:
+            raise AdapterError(str(e))
+
+    def test_connection(self) -> bool:
```
Comment (harini-venkataraman) [inline src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py:68]: @hari-kuriakose A QQ here - 
Do we need to add a sleep with the configured seconds here just to confirm the sleep happens right in test connection, or should we confine it just to the core methods like completion etc,
image
The behavior right now is, returns True immediately, even if the wait is configured. Felt, this might sometimes mislead while testing. But in another way, addition of sleep is not any functionality of this adapter. 
Please advice with your inputs here.

Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py:68]: @hari-kuriakose Even though it is ok to return immediately in test connection, adding sleep will be better for a consistent UX. Let's do this.

File: src/unstract/sdk/adapters/llm/no_op_llm/README.md
Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/llm/no_op_llm/README.md:1]: Done.

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json (hunk 17 above)
Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/static/json_schema.json]: Done.

File: src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py (hunk 19 above)
Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py]: Done.

File: src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py (hunk 21 above)
Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/x2text/no_op/src/no_op_x2text_adapter.py]: Done.

File: src/unstract/sdk/adapters/embedding/no_op_embedding/src/no_op_embedding_adapter.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply src/unstract/sdk/adapters/embedding/no_op_embedding/src/no_op_embedding_adapter.py]: @chandrasekharan-zipstack Reconsidering this: https://github.com/Zipstack/unstract-sdk/pull/109/files#r1804070938

**Comment by hari-kuriakose** [comment]
> LGTM overall. Have 2 minor concerns around checking for no-op type in places where adapters are used - I hope there's general consensus around these @hari-kuriakose @harini-venkataraman @gaya3-zipstack @Deepak-Kesavan
> 
>     1. [Checking it in index before adding nodes to vectorDB](https://github.com/Zipstack/unstract-sdk/pull/109/files#r1796155556)
> 
>     2. Adding a method to check for this instance in the base class - related to table extractor

@chandrasekharan-zipstack 

1. Please find https://github.com/Zipstack/unstract-sdk/pull/109#discussion_r1802740388

**Review by hari-kuriakose — APPROVED** [review]
@harini-venkataraman LGTM overall.
